include/libvfio-user.h

```diff
@@ -362,9 +369,37 @@ typedef enum vfu_reset_type {
  *      vfu_device_quiesced returns the device is no longer quiesced.
  *
  * A quiesced device should expect for any of the following callbacks to be
- * executed:
- * vfu_dma_register_cb_t, vfu_unregister_cb_t, vfu_reset_cb_t, and transition.
- * These callbacks are only called after the device has been quiesced.
+ * executed: vfu_dma_register_cb_t, vfu_unregister_cb_t, vfu_reset_cb_t, and
+ * the migration transition callback. These callbacks are only called after the
+ * device has been quiesced.
+ *
+ * The following example demonstrates how a device can use vfu_map_sg and
+ * friends while quiesced:
+ *
+ * A DMA region is mapped, libvfio-user calls the quiesce callback but the
+ * cannot immediately quiesce:
+ *
+ *     int quiesce_cb(vfu_ctx_t *vfu_ctx) {
+ *         errno = EBUSY;
+ *         return -1;
+ *     }
+ *
+ * While quiescing, the device can continue operate as normal, including
+ * calling functions such as vfu_map_sg. Then, the device finishes quiescing:
+ *
+ *  vfu_quiesce_done(vfu_ctx, 0);
+ *
+ * From this point onwards the device cannot call functions such as vfu_map_sg.
```

Comment:
I think it's more accurate to say something like:

```
At this point, the device must have stopped using functions like vfu_map_sg(),
for example by pausing any I/O threads.
```

Because the callbacks do allow that again immediately.

Reply:
ok